test/e2e/locators/recordset.ts

```diff
@@ -161,6 +165,68 @@ export default class RecordsetLocators {
     return container.locator('.side-panel-resizable');
   }
 
+  static getFacetById(container: Page | Locator, idx: number): Locator {
+    return container.locator(`.fc-${idx}`);
+  }
+
+  static getFacetHeaderButtonById(facet: Locator, idx: number): Locator {
+    return facet.locator(`.fc-heading-${idx} button`);
+  }
+
+  // get child of accordion group, sibling to accordion heading
+  static getFacetCollapse(facet: Locator): Locator {
+    return facet.locator('.accordion-collapse');
+  }
+
+  static getFacetOptions(facet: Locator): Locator {
+    return facet.locator('.chaise-checkbox label')
+  }
+
+  static getFacetOption(facet: Locator, optionIdx: number) {
+    return facet.locator(`.checkbox-${optionIdx}`);
+}
 
+  /* range facet selectors */
+  // there's integer/float/date/timestamp inputs
+  static getRangeInput(facet: Locator, className: string): Locator {
```

Comment:
I think we should improve these functions instead just rewriting them in playwright. Passing the actual class from the test completely defeats the purpose of these functions. Our goal is to keep all the implementation details hidden in these functions. So if we want to change a class/implementation we would just change these functions and not the actual specs.

You could do something similar to what we did for [here](https://github.com/informatics-isi-edu/chaise/blob/master/test/e2e/locators/recordedit.ts#L143)). So,

```ts
static getFacetRangeInputs(facet:Locator, isTimestamp?: boolean) : Locator {
  return {
    minInput: facet.locator(...),
    maxInput: facet.locator(...),
    minClear: ..,
    maxClear: ...,
    ...
  }
}
```